### DISCUSSION TITLE
How to add an new asset wich will be take by HtmlWebpackPlugin

### DISCUSSION THREAD

**Discussion by ghost** [webpack/webpack#18339] [Q&A]
Hi,
I'm trying to develop a Webpack5 plugin which make change in assets and can add new assets.
To add new asset, I'm using `compilation.emitAsset()`.

New assets are correctly created (I can show new JS files) but  HtmlWebpackPlugin which is in the list of plugins doesn't push them into the html file.

How can I create new assets which will be treated by HtmlWebpackPlugin ?

Thanks and regards

**Comment by alexander-akait** [comment]
Which version of `html-webpack-plugin` do you use? Can you create reproducible test repo? Also please try to disable the `cache` option

**Reply by ghost** [reply]
html-webpack-plugin V5.6.0
webpack V5.91.0
To test my case, I've update the example plugin `FileListPlugin.cjs`:
```js
class FileListPlugin {
    static defaultOptions = { outputFile: 'assets.js',};

    constructor(options = {}) {
        this.options = { ...FileListPlugin.defaultOptions, ...options };
    }

    apply(compiler) {
        const pluginName = FileListPlugin.name;
        const { webpack } = compiler;
        const { Compilation } = webpack;
        const { RawSource } = webpack.sources;

        compiler.hooks.thisCompilation.tap(pluginName, (compilation) => {
            compilation.hooks.processAssets.tap(
                {
                    name: pluginName,
                    stage: Compilation.PROCESS_ASSETS_STAGE_SUMMARIZE,
                },
                (assets) => {
                    const content =  `console.log(${Object.keys(assets).map((asset) => `'${asset}'`)});`;
                    compilation.emitAsset(
                        this.options.outputFile,
                        new RawSource(content)
                    );
                }
            );
        });
    }
}
module.exports = { FileListPlugin };
```
In webpack.config.js:
```js
...
    plugins: [
        new FileListPlugin(),
        new CleanWebpackPlugin(),
        new HtmlWebpackPlugin(),
    ]
...
```

As result, I obtain index.html, main.js and assets.js, but only main.js is added into indes.html.

**Reply by alexander-akait** [reply]
Do you want to include this file into HTML, if yes, please use the template option and include file file into your HTML file, because by default htm-webpack-plugin include only chunks, not all assets
